[ADD][15.0] `purchase_sale_inter_company` without dependency on `stock`

yankinmax opened the pull request:
This module purpose is to split original `purchase_sale_inter_company` **to remove dependency on `stock`**

chafique-delli committed:
[ADD] purchase_sale_inter_company module

chafique-delli committed:
add warehouse_id in the values for create sale order

chafique-delli committed:
refactor code after add dependency on account_invoice_inter_company

chafique-delli committed:
check product is intercompany

chafique-delli committed:
extend invoice_method field and add action_cancel method for purchase and auto_purchase_line_id field in sale_order_line

chafique-delli committed:
rename inter_company_po_to_so_view.xml to res_company_view.xml and add purchase_order_view.xml

chafique-delli committed:
use product_id_change_with_wh method for return sale line data

chafique-delli committed:
fix README file

chafique-delli committed:
remove inter company user

chafique-delli committed:
remove unnecessary dependencies in openerp.py and update of the unit price in purchase line when validating the sale order

chafique-delli committed:
add demo data and tests

chafique-delli committed:
fix partner_shipping_id in _prepare_sale_order_data method

chafique-delli committed:
add force_company in context when create inter company sale oder

sebastienbeau committed:
[FIX] fix setting the force company pass the cest_company as an id in order to load it with the correct context

chafique-delli committed:
replace sale_line_vals by sale_line_data

chafique-delli committed:
fix tests

chafique-delli committed:
fix following @elicoidal's comments and add french translation

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

chafique-delli committed:
fix domain for search users in _check_intercompany_product method

chafique-delli committed:
fix multi-company access rights issue

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[MIG] purchase_sale_inter_company: Migration to 11.0 (from 8.0)

carlosdauden committed:
[IMP] purchase_sale_inter_company: Recover intercompany user

pedrobaeza committed:
[FIX] purchase_sale_inter_company: Transfer dropship pickings correctly

carlosdauden committed:
[IMP] purchase_sale_inter_company: Better relation between docs. Translate.

yajo committed:
[FIX] purchase_sale_inter_company: No buggy demo data

Make tests compatible with https://github.com/OCA/multi-company/pull/110, for the same reason as the specified in that link.

chienandalu committed:
[FIX] tests and other issues

mourad-ehm committed:
[MIG] purchase_sale_inter_company: Migration to 12.0

Commit:
Update purchase_sale_inter_company/README.rst OCA Review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA Review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

Commit:
Update purchase_sale_inter_company/README.rst OCA review

Co-Authored-By: mourad-ehm <mourad.elhadj.mimoune@akretion.com>

mourad-ehm committed:
[FIX] OCA Review

mourad-ehm committed:
[FIX] ref uom.product_uom_unit

mourad-ehm committed:
[FIX] import Warning

mourad-ehm committed:
[FIX] import Warning (To squash)

mourad-ehm committed:
[FIX] demo test data

mourad-ehm committed:
[FIX] add demo product

mourad-ehm committed:
[FIX] product_qty

mourad-ehm committed:
[FIX] unit test

mourad-ehm committed:
[FIX] pylint

mourad-ehm committed:
[FIX] set readonly = false for  all res_config_settings fields

bealdav committed:
IMP purch_sale_inter_cpny: layout settings

oca-travis committed:
[UPD] Update purchase_sale_inter_company.pot

eLBati committed:
Added translation using Weblate (Italian)

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (25 of 25 strings)

Translation: multi-company-12.0/multi-company-12.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-purchase_sale_inter_company/it/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-12.0/multi-company-12.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-purchase_sale_inter_company/

AdriaGForgeFlow committed:
[IMP] purchase_sale_inter_company: black, isort

AdriaGForgeFlow committed:
[13.0][MIG] purchase_sale_inter_company

rename intercompany_user_id to intercompany_sale_user_id for compatibility with account_invoice_inter_company

[IMP] res_config_settings view

oca-travis committed:
[UPD] Update purchase_sale_inter_company.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/

dong-z committed:
Added translation using Weblate (Chinese (Simplified))

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 14.8% (4 of 27 strings)

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/zh_CN/

AaronHForgeFlow committed:
[FIX]purchase_sale_inter_company tests
* product multi company has company because of the default method

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/

AaronHForgeFlow committed:
[IMP] purchase_sale_inter_company
* move demo data to SetupClass

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.1

chienandalu committed:
[FIX] purchase_sale_inter_company: commitment_date

The field was renamed from `requested_date` to `commitment_date`

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.2

Commit:
[MIG] purchase_sale_inter_company: Migration to 14.0

enriquemartin committed:
[FIX] purchase_sale_inter_company: propagate correct uom to sale.order.line

Commit:
[13.0] formatting

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (27 of 27 strings)

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/es/

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.4

kittiu committed:
[13.0][FIX] purchase_sale_inter_company, PO->SO error when has line_section

OCA-git-bot committed:
purchase_sale_inter_company 13.0.1.0.5

Commit:
[IMP] purchase_sale_inter_company: black, isort, prettier

oca-travis committed:
[UPD] Update purchase_sale_inter_company.pot

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

Kev-Roche committed:
[FIX] purchase_sale_inter_company: fix new sale date_order

OCA-git-bot committed:
purchase_sale_inter_company 14.0.1.0.1

DhavalTforgeFlow committed:
[IMP] purchase_sale_inter_company: black, isort, prettier

DhavalTforgeFlow committed:
[MIG] Migration to version 15.0

DhavalTforgeFlow committed:
[MIG] purchase_sale_inter_company: Migration to 15.0

yankinmax committed:
[ADD] purchase_sale_inter_company:

This is the splitted module from original to remove dependency on stock

yankinmax commented on purchase_sale_inter_company/models/purchase_order.py:
```diff
@@ -103,30 +103,25 @@ def _inter_company_create_sale_order(self, dest_company):
+                "partner_id": dest_company.partner_id.id,
                 "date_order": self.date_approve,
                 "auto_purchase_order_id": self.id,
             }
```
@gurneyalex can you pls take a look, I have doubts about it.
1. `delivery_address = direct_delivery_address or dest_company.partner_id` looks correct to me as we specify destination address on new generated SO
2. `partner_id` on new SO changed to destination company partner. On Invoice creation `Incompatible companies on records` is raised if partner is the same as on the PO. So, for example we got Invoice with company B and partner with company A.

yankinmax commented:
@gurneyalex @DhavalTforgeFlow 
The error in tests is weird.
```
Draft Invoice INV/2022/00001 (P00009)' belongs to company 'Company B' and 'Partner' (partner_id: 'Company A') belongs to another company
```
We need to generate SO (Company B) from PO (Company A) with `partner_id` who is `partner_id` of (Company A).
Then Invoice is created from generated SO (Company B). Seems legit.
Should we modify `partner_id` of the Invoice to become (Company B) during invoice creation?

gurneyalex commented:
/ocabot merge minor

gurneyalex commented:
oh shucks wrong pr

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

Pricelist currency should be same with PO/SO document
If corresponding pricelist currency is not the same find or create propriate pricelist

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

Company contacts can be Vendors to create SO from PO

gurneyalex reviewed (changes requested):
Can you check the failing test: https://github.com/OCA/multi-company/actions/runs/3363433913/jobs/5576569084#step:8:174

yankinmax commented:
@gurneyalex I couldn't reproduce the error locally. The test is red since PR is opened.
The error message is: `Draft Invoice INV/2022/00001 (P00009)' belongs to company 'Company B' and 'Partner' (partner_id: 'Company A')`, but the invoice and partner are from the same company in tests. I'll try one more time

DhavalTforgeFlow commented:
Can you please change the description in README.rst file? It seems like we have old description of configurations when settings were there under Setting --> Companies --> Company. Now it's under general settings (res.config.settings).

yankinmax commented:
@DhavalTforgeFlow updated with proper README

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

If purchase_exception or sale_exception modules are installed
we have an issue in tests, because order's confirm methods won't be called.
Possible fix is to set ignore_exception=True on orders in tests.
Exception check could be skipped by setting check_exception=False,
as mentioned in base_exception, but in fact it doesn't work.

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

Pricelist is set with the company

JasminSForgeFlow commented:
@yankinmax can you please provide me access to push changes on this PR as well as the purchase_sale_stock_inter_company PR also?

As purchase_sale_inter_company merged in v14 has changes in code and tests for fixing issues related to multi-company in the v14 test.

Thanks

yankinmax commented:
@JasminSForgeFlow sent invitation, thanks

JasminSForgeFlow committed:
[15.0][IMP] purchase_sale_inter_company:
Update v14 latest changes related to test case failing

mariadforgeflow reviewed (approved):
LGTM! Code and functional review:)

yankinmax commented:
@JasminSForgeFlow @mariadforgeflow thanks for the work done

AaronHForgeFlow commented:
/ocabot merge nobump

mariadforgeflow commented:
Hi @gurneyalex! Now all tests have passed, could you merge it? Thanks!

gurneyalex commented:
/ocabot merge nobump
